### DISCUSSION TITLE
Reparenting a proxyshape?

### DISCUSSION THREAD

**Discussion by pierre-ed** [Autodesk/maya-usd#2273] [Q&A] [answered]
Hi everyone.

Here I go again :) with yet another painfully simple question: is reparenting a proxyshape under another transform supported?

My current testing scenario is pretty straightforward: I load a `mayaUsdProxyShape` using `Create->USD->Stage From File`, and then reparent the created proxyshape under an existing transform using `cmds.parent(proxyshape, new_parent, shape=True, relative=True)`. This operation triggers no error, but the proxyshape vanishes from the VP2 renderer (even after a call to `cmds.refresh(force=True)`).   

Internally, computations seem correct however, since my proxyshape reappears with the expected transform when I switch renderer to the GL Hydra render delegate.

![20220411__maya__reparent2](https://user-images.githubusercontent.com/5444972/162754627-fd4deed6-af24-4e2e-a384-3342e91d9c22.gif)

Am I missing something? Is there a workaround to force the refreshing of the scene or render delegate behind the vp2RenderDelegate? 

Thanks in advance for your insight.
-- Pierre

#### Specs

- OS & version: CentOS Linux release 7.6.1810 (Core)
- Compiler & version: gcc (GCC) 6.3.1 20170216
- Maya version: Maya 2022.3
- Maya USD: 0.15.0
- Pixar USD: 21.11

**Comment by williamkrick** [comment, marked as answer]
Hi Pierre,

I believe this is a Maya bug with re-parenting shapes that have MPxSubSceneOverride's attached. If you run an `ogs -reset` the objects should re-appear. @santosd I can reproduce this one locally, will you convert to an issue something internally for us? As a work around re-parenting the transform above the proxyShape should work fine.

William

**Reply by pierre-ed** [reply]
Thank you, William.   
The viewport refreshing command you mentioned works like a charm. :)

**Reply by williamkrick** [reply]
Great! A word of warning, `ogs -reset` is quite slow for complicated scenes. So if you do find more places you need to use it for correct behavior please keep reporting them.
